### DISCUSSION TITLE
Dynamically create reverse-aliases for email forwarding from email address pattern

### DISCUSSION THREAD

**Discussion by andreabenetton** [simple-login/app#2019] [Feature Requests]
Imagine a scenario where an organization utilizes a custom domain to manage its email communications. Within this domain, different users or departments have specific email addresses. However, for security and privacy reasons, the organization prefers not to grant these users direct access to the email management system's web interface, such as SimpleLogin.

I'm proposing a solution that addresses a specific need: the ability to dynamically create reverse-aliases for email forwarding based on certain conditions. This process would be automatic, requiring no direct intervention from the end users. The system would recognize emails coming from authorized senders and, based on the recipient's address format (which follows a predefined pattern), generate a reverse-alias on the fly.

An email sent to andrea.benetton--blueteam.ee@customdomain.com would trigger the system if Andrea Benetton is an authorized sender. The system recognizes the pattern in the recipient's address, understands it's meant for andrea.benetton@blueteam.ee and creates or utilizes an existing alias to forward the email accordingly. This operation ensures that emails are routed correctly without exposing the backend processes or requiring manual alias setup by the users.

(Zendesk request 2640264)

**Comment by thisislawatts** [comment]
Love this idea @andreabenetton, thanks for raising here. Matches a very real need for a small community project we are running where there is a need for a unified email domain without the full per/user pricing of standard email providers.

**Comment by eopo** [comment]
Building upon that idea:
If SL is set as upstream SMTP-Server, it can decide on split horizon to directly relay internal mail to the real internal Mailserver and modify outgoing mail by creating an alias if not existing and changing the `From` header.

This way, an internal user can send Mails to the real recipient mail address whilst not having to care about the alias at all.
